Ngclient docstrings improvement

ivanayov opened the pull request:
This change unifies ngclient docstrings language, including wording and quoting. No major changes included, as docs look well-formed.

Fixes #1844

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [n/a] Tests have been added for the bug fix or new feature
- [n/a] Docs have been added for the bug fix or new feature

ivanayov committed:
Unify quoting in ngclient docstrings

This change updates the docstrings library of ngclient with
unified double backtick quoting for better readability

Signed-off-by: Ivana Atanasova <iyovcheva@vmware.com>

ivanayov committed:
Unify article in ngclient docstrings

This change updates the docstrings library of ngclient with no
article for all Args in order to be unified amongst all python-tuf
docstrings

Signed-off-by: Ivana Atanasova <iyovcheva@vmware.com>

ivanayov committed:
Unify wording of docstrings language in ngclient

This change unifies common wording in the docstrings library of
ngclient, like "Args" vs. "Arguments"

Signed-off-by: Ivana Atanasova <iyovcheva@vmware.com>

lukpueh commented on tuf/ngclient/_internal/trusted_metadata_set.py:
```diff
@@ -3,17 +3,17 @@
 Loaded metadata can be accessed via index access with rolename as key
-(trusted_set[Root.type]) or, in the case of top-level metadata, using the helper
-properties (trusted_set.root).
+(``trusted_set[Root.type]``) or, in the case of top-level metadata, using the helper
```
line length

lukpueh commented on tuf/ngclient/_internal/trusted_metadata_set.py:
```diff
@@ -99,55 +99,55 @@ def __init__(self, root_data: bytes):
 
     def __iter__(self) -> Iterator[Metadata]:
-        """Returns iterator over all Metadata objects in TrustedMetadataSet"""
+        """Returns iterator over all ``Metadata`` objects in ``TrustedMetadataSet``"""
```
line length

lukpueh commented on tuf/ngclient/_internal/trusted_metadata_set.py:
```diff
@@ -256,32 +256,32 @@ def _check_final_timestamp(self) -> None:
+        """Verifies and loads ``data`` as new snapshot metadata.
 
         Note that an intermediate snapshot is allowed to be expired and version
         is allowed to not match timestamp meta version: TrustedMetadataSet will
```
```suggestion
        is allowed to not match timestamp meta version: ``TrustedMetadataSet`` will
```
--> line length

lukpueh commented on tuf/ngclient/_internal/requests_fetcher.py:
```diff
@@ -22,11 +22,11 @@
 class RequestsFetcher(FetcherInterface):
-    """A concrete implementation of FetcherInterface based on the Requests
+    """A concrete implementation of ``FetcherInterface`` based on the Requests
     library.
```
Minor nit: According to our style guide, this should fit on one line. What about...
```suggestion
    """An implementation of ``FetcherInterface`` based on the requests library.
```
?

lukpueh commented on tuf/ngclient/fetcher.py:
```diff
@@ -72,21 +73,21 @@ def fetch(self, url: str) -> Iterator[bytes]:
-        """Opens a connection to 'url' and downloads the content
-        up to 'max_length'.
+        """Opens a connection to ``url`` and downloads the content
+        up to ``max_length``.
```
Minor nit: According to our style guide, this should fit on one line. Also, at first glance this could be read as if hitting max length would just stop the download, although it actually raises an exception, which is described below anyway. Maybe we can just say as we do in `download_bytes`?
```suggestion
        """Download file from given url.
```
?

lukpueh replied:
Optional: If we do want to give any more info here, e.g. in a paragraph after the opening line, I'd probably mention that `download_file` should be used in a `with` block, to always release the allocated file resources even if download fails. (not a blocker for this PR).

lukpueh replied:
Minor nit 2: Consider consistently wrapping `url` in double backticks in opening lines of all functions of this module

lukpueh commented on tuf/ngclient/updater.py:
```diff
@@ -57,23 +57,23 @@
     Args:
         metadata_dir: Local metadata directory. Directory must be
-            writable and it must contain a trusted root.json file.
+            writable and it must contain a trusted ``root.json`` file
```
Not sure if we need to backtick-wrap file names? Maybe just quote...?
```suggestion
            writable and it must contain a trusted 'root.json' file
```

lukpueh reviewed (changes requested):
Thanks a lot for the patch, @ivanayov. I left a few minor comments inline...

ivanayov committed:
Fix line lengths in ngclient docstrings

This change shortens line lengths that exceed the requiremets and
adds more clarification on methods where the short message is not
complete enough

Signed-off-by: Ivana Atanasova <iyovcheva@vmware.com>

ivanayov commented:
Thank you @lukpueh ! Just pushed an update addressing your comments.

lukpueh commented:
Thanks for the quick updates, @ivanayov! Merging...
